.github/workflows/cypress.yml

```diff
@@ -0,0 +1,79 @@
+name: E2E test
+
+on:
+  push:
+    branches:
+      - develop
+      - trunk
+  pull_request:
+    branches:
+      - develop
+
+jobs:
+  build:
+    uses: 10up/embed-block-figma/.github/workflows/build-release-zip.yml@develop
+  cypress:
+    needs: build
+    name: ${{ matrix.core.name }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        core:
+          - {name: 'WP latest', version: 'latest'}
+          - {name: 'WP minimum', version: 'WordPress/WordPress#5.7'}
+          - {name: 'WP trunk', version: 'WordPress/WordPress#master'}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Download build zip
+      uses: actions/download-artifact@v4
+      with:
+        name: ${{ github.event.repository.name }}
+        path: ${{ github.event.repository.name }}
+    - name: Display structure of downloaded files
+      run: ls -R
+      working-directory: ${{ github.event.repository.name }}
+    - name: Cache node_modules
```

Comment:
Probably want a step before this where we install the proper node version, see https://github.com/10up/insecure-content-warning/blob/develop/.github/workflows/cypress.yml#L39